### PR TITLE
testing create user

### PR DESCRIPTION
**Pull request by mvn-haphan-dn**
## Description
Unitest create user
## Reference Requirements
- Task ID: TI-29

## Migrations
YES
Create migartion update_default_image_users_table to set default image for user

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

**Comment by at-haytran** [comment]
LGTM

File: app/Http/Controllers/Admin/UserController.php [hunk 1]
```diff
@@ -103,7 +103,7 @@ public function store(CreateUserRequest $request)
             $user->save();
             $image->move($destinationPath, $nameNew);
         } else {
-            $user->avatar = $request->avatar;
+            $user->avatar = 'default-user.png';
```
Comment (at-vietpham) [inline app/Http/Controllers/Admin/UserController.php]: `default-user.png` nên được define trong config hoặc Model User

File: tests/Browser/Pages/Backend/CreateUserTest.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+     *
+     * @param string $name    name of field
+     * @param string $content content
+     * @param string $message message show when validate
+     *
+     * @dataProvider listCaseTestValidateForInput
+     *
+     * @return void
+     */
+    public function testValidateForInput($name, $content,$message)
```
Comment (at-vietpham) [inline tests/Browser/Pages/Backend/CreateUserTest.php]: Thêm dấu cách ` `  trước `$message`.

File: tests/Browser/Pages/Backend/CreateUserTest.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            ['identity_number', '', 'The identity number field is required.'],
+            ['dob', '', 'The dob does not match the format Y-m-d.'],
+            ['address', '', 'The address must be a string.'],
+        ];
+    }
+
+    /**
+     * Dusk test validate for input
+     *
+     * @param string $name    name of field
```
Comment (at-vietpham) [inline tests/Browser/Pages/Backend/CreateUserTest.php]: CI ko check mấy file trong tests nhưng cũng cần đúng convention. Căn thẳng hàng phần comment cho variable

File: tests/Browser/Pages/Backend/CreateUserTest.php [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+     *
+     * @dataProvider listCaseTestValidateForInput
+     *
+     * @return void
+     */
+    public function testValidateForInput($name, $content,$message)
+    {
+        $this->browse(function (Browser $browser) use ($name, $content, $message) {
+            $browser->visit('admin/users/create')
+                ->type('name', '')
```
Comment (at-vietpham) [inline tests/Browser/Pages/Backend/CreateUserTest.php]: Nếu không input gì thì có cần `->type()` không?

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by at-vietpham** [comment]
LGTM 🎃

**Pull request merged**